### ISSUE TITLE
Search Error : Uncaught (in promise) TypeError: t.toLowerCase is not a function

### ISSUE THREAD

**Issue by jnan88** [squidfunk/mkdocs-material#4533]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [ ] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [ ] ... the documentation does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem

### Description

run with docker squidfunk/mkdocs-material:8.5.6

docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material:8.5.6

unable use search in jieba

### Expected behaviour

lunr.ja.min.js:1 
       Uncaught (in promise) TypeError: t.toLowerCase is not a function
    at lunr.ja.min.js:1:862
    at Array.map (<anonymous>)
    at e.ja.tokenizer (lunr.ja.min.js:1:823)
    at t.Builder.add (lunr.js:2479:23)
    at t.Builder.<anonymous> (index.ts:217:14)
    at t (lunr.js:53:10)
    at new U (index.ts:177:18)
    at index.ts:146:15
    at Generator.next (<anonymous>)
    at s (search.16e2a7d4.min.js:1:953)

### Actual behaviour

unable use search 

### Steps to reproduce

1. run docker 
2. open index
3. show Console

### Package versions

- Python: `python --version`
- MkDocs: `mkdocs --version`
- Material: `pip show mkdocs-material | grep -E ^Version`

### Configuration

```yaml
site_name: mkdocs知识站点
theme: 
  name: material #mkdocs #readthedocs
  language: zh
  highlightjs: true
  hljs_languages:
    - yaml
    - rust
  features:
    - navigation.instant
    - navigation.tracking
    - navigation.sections
    - navigation.expand
    - navigation.prune
    - navigation.indexes
    - navigation.tabs
    - navigation.tabs.sticky
    - navigation.top
    - toc.follow
    - toc.integrate
    - content.tabs.link
    #- search.suggest
    #- search.highlight
  plugins:
    - tags:
      tags_file: tags.md
    # - search:
    #   lang: 
    #     - en
    #     - zh
    #   jieba_dict: dict.txt
    #   separator: '[\s\-,:!=\[\]()"/]+|\.(?!\d)|&[lg]t;|(?!\b)(?=[A-Z][a-z])'

markdown_extensions:
  - tables
  - attr_list
  - md_in_html
  - def_list
  - admonition
  - pymdownx.tasklist:
      custom_checkbox: true
  - pymdownx.tabbed:
      alternate_style: true
  - pymdownx.highlight:
      linenums: true
      anchor_linenums: true
  - pymdownx.inlinehilite
  - pymdownx.snippets
  - pymdownx.details
  - pymdownx.superfences
  - pymdownx.critic
  - pymdownx.caret
  - pymdownx.keys
  - pymdownx.mark
  - pymdownx.tilde

nav:
    - Home: index.md
    - 关于: about.md
    - 示例:
      - material 主题示例: example/note.md
```

### System information

- MacOS Monterey 12.6
- Microsoft Edge 版本 106.0.1370.47 (正式版本) (x86_64)

**Comment by squidfunk** [comment]
Thanks for reporting. Duplicate of #4469 and #4069. This issue is fixed in Insiders. Please see https://github.com/squidfunk/mkdocs-material/issues/4069#issuecomment-1172851670 why backporting the fix is not economical.

**Comment by jnan88** [comment]
How to solve this problem ，Can a new version be released？

**Comment by squidfunk** [comment]
The problem will be fixed as soon as the [next funding goal](https://squidfunk.github.io/mkdocs-material/insiders/#10000-carolina-reaper) is hit, which will ship the new search. If you need it before, you would need to [become a sponsor](https://squidfunk.github.io/mkdocs-material/insiders/#how-to-become-a-sponsor).

**Comment by jnan88** [comment]
> The problem will be fixed as soon as the [next funding goal](https://squidfunk.github.io/mkdocs-material/insiders/#10000-carolina-reaper) is hit, which will ship the new search. If you need it before, you would need to [become a sponsor](https://squidfunk.github.io/mkdocs-material/insiders/#how-to-become-a-sponsor).

thinks

**Comment by squidfunk** [comment]
The fix was released in 9.0.0.
